File: tests/test_cpg/fixtures/parity_crypto.py
"""Cross-language crypto weakness fixture — Python."""

import hashlib
from flask import Flask, request

app = Flask(__name__)


@app.route("/register", methods=["POST"])
def register():
    password = request.form.get("password", "")  # $ source=crypto_weakness
    h = hashlib.md5(password.encode()).hexdigest()  # $ sink=crypto_weakness
    return f"Hash: {h}"


@app.route("/hash_data")
def hash_data():
    data = request.args.get("data", "")  # $ source=crypto_weakness
    h = hashlib.sha1(data.encode()).hexdigest()  # $ sink=crypto_weakness
    return h

File: tests/test_cpg/fixtures/parity_crypto.js
/** Cross-language crypto weakness fixture — JavaScript (Node.js). */

var crypto = require('crypto');

function hashPassword(req, res) {
    var password = req.body.password;  // $ source=crypto_weakness
    var hash = crypto.createHash('md5').update(password).digest('hex');  // $ sink=crypto_weakness
    res.send('Hash: ' + hash);
}

function hashData(req, res) {
    var data = req.query.data;  // $ source=crypto_weakness
    var hash = crypto.createHash('sha1').update(data).digest('hex');  // $ sink=crypto_weakness
    res.send(hash);
}
